### DISCUSSION TITLE
Get [node] from [dblclick] event

### DISCUSSION THREAD

**Discussion by TeoDar** [primefaces/community#445] [PrimeVue] [answered]
Good day, PrimeVue team.
Could you please suggest how I can get a **node** on a double click event on a Tree element so that I can expand it later? I tried to add a _:pt_ handler to the _content_ container, but I don’t know what to do next with the **event** and how to get the **node** through it?(

Another question. Why doesn't my event.preventDefault() prevent text from being highlighted in the browser?

[Here is an example](https://stackblitz.com/edit/ucv6bt?file=src%2FApp.vue) forked from a basic Tree component example.

**Comment by tugcekucukoglu** [comment]
`expandedKeys` property controls the expansion state and when double clicked an item, expandedKeys can be updated.

**Reply by TeoDar** [reply]
I understand that, but you don't answer my questions(

**Comment by Rakasch** [comment, marked as answer]
If you add a plain javascript event handler, you will not get any vue related data.

You could either workaround it using the selectedNode:
https://stackblitz.com/edit/ucv6bt-9yub2h?file=src%2FApp.vue

Or you could go the plain js way and add a data-key="xy" to all your nodes, then get it in the doubleclick by event.target.dataset.key;

**Reply by TeoDar** [reply]
Tricky solution. Thanks a lot!
Although, I was hoping that there was some way to determine the node by the event that was called)
